tutorials/images/Images_Tutorial.ipynb

```diff
@@ -0,0 +1,444 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "L_names = [LF_street,LF_vehicles,LF_distance,LF_size,LF_number]\n",
+    "\n",
+    "L = np.zeros((len(L_names),loader.train_num)).astype(int)\n",
+    "for i in range(loader.train_num):\n",
+    "    L[0,i] = LF_street(P.primitive_mtx[i,0],P.primitive_mtx[i,1])\n",
+    "    L[1,i] = L_names[1](P.primitive_mtx[i,0],P.primitive_mtx[i,2])\n",
+    "    L[2,i] = L_names[2](P.primitive_mtx[i,0],P.primitive_mtx[i,3], P.primitive_mtx[i,4])\n",
+    "    L[3,i] = L_names[3](P.primitive_mtx[i,0],P.primitive_mtx[i,3], P.primitive_mtx[i,5])\n",
+    "    L[4,i] = L_names[4](P.primitive_mtx[i,0],P.primitive_mtx[i,3], P.primitive_mtx[i,6])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Calculate and Show Accuracy and Coverage of Heuristic Functions**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "total = float(loader.train_num)\n",
+    "\n",
+    "stats_table = np.zeros((5,2))\n",
+    "for i in range(5):\n",
+    "    stats_table[i,1] = np.sum(L[i,:] == loader.train_ground)/float(np.sum(L[i,:] != 0))\n",
+    "    stats_table[i,0] = np.sum(L[i,:] != 0)/total"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "stats_table = pd.DataFrame(stats_table, index = [i.__name__ for i in L_names], columns = [\"Coverage\", \"Accuracy\"])\n",
+    "stats_table"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate Training Set"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from snorkel.learning import GenerativeModel\n",
+    "from scipy import sparse\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "L_train = sparse.csr_matrix(L.T)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Majority Vote**\n",
+    "\n",
+    "To get a sense of how well our heuristic functions are, we calcuate the accuracy of the training set labels if we took the majority vote label for each data point."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mv_labels = np.sign(np.sum(L.T,1))\n",
+    "print ('Coverage of Majority Vote on Train Set: ', np.sum(np.sign(np.sum(np.abs(L.T),1)) != 0)/float(loader.train_num))\n",
+    "print ('Accuracy of Majority Vote on Train Set: ', np.sum(mv_labels == loader.train_ground)/float(loader.train_num))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Generative Model**\n",
+    "We assume that the heuristic functions are conditionally independent of the true label and train the generative model using the labels assigned by the heuristic functions"
```

Comment:
Conditionally independent *given* the true label

Reply:
+ Here might be a good place to put a pointer in to Coral @paroma e.g. "but, to see more advanced stuff we can do, see..."